axum/CHANGELOG.md

```diff
@@ -12,6 +12,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   Use `axum::middleware::from_extractor` instead ([#1077])
 - **breaking:** Allow `Error: Into<Infallible>` for `Route::{layer, route_layer}` ([#924])
 - **breaking:** `MethodRouter` now panics on overlapping routes ([#1102])
+- **breaking:** Nested `Router`s will no longer delegate to the outer `Router`'s
+  fallback. Instead you must explicitly set a fallback on the inner `Router` ([#1086])
```

Comment:
Oh, we actually had this previously? That makes it a little more unfortunate that we now don't support it. Why was it previously possible without a `nest` / `nest_service` split, but is no longer (or is it just removed so `nest` doesn't do magic `Router` special casing)?

Reply:
> or is it just removed so nest doesn't do magic Router special casing

yeah its because of this. We had it previously but only if you passed a `Router` to `nest`. It wasn't documented either and quite magical I think. Thats what I was getting at [here](https://github.com/tokio-rs/axum/pull/1086#issuecomment-1195323750). I agree its unfortunate but think its the right move overall.